SirMorland/huld-hub
# chore: strapi share db to frontend devs

## Description
- added a way to share db on app startup
- added some more info on readme


## Considerations and implementation
- next thing is to have proper content types, then generate some placeholders and add a mysql dump so frontend dev can start devving


### How to test


Generate some content for the types that your local has. Then follow the instructions:


- Make sure `mysql` container is up and running
- Make sure you command line is located at `./backend`
- Run `docker ps` to get the container ID of the mysql `container`:
```bash
docker ps                        
CONTAINER ID   IMAGE           COMMAND                  CREATED         STATUS         PORTS                                       NAMES
2b8f793f9429   strapi/strapi   "docker-entrypoint.s…"   2 minutes ago   Up 2 minutes   0.0.0.0:1337->1337/tcp, :::1337->1337/tcp   huldhub_backend
ca35896ea2b7   mysql           "docker-entrypoint.s…"   2 minutes ago   Up 2 minutes   3306/tcp, 33060/tcp                         huldhub_database
```
- Run `docker exec ${MSQYL_CONTAINER_ID} /usr/bin/mysqldump -u root --password=strapi strapi > ./mysql-dump/backup.sql`. For example:
```bash
docker exec ca35896ea2b7 /usr/bin/mysqldump -u root --password=strapi strapi > ./mysql-dump/backup.sql
```
- Remove `./backend/data` (we don't need this anymore)
- Shut down the all the containers with `docker-compose down`
- Remove all containers with `docker-compose rm` or then `docker container prune` to for a fresh docker-compose build and setup.
- Then run `docker-compose up`. 
- And voila, all previous db is now available according to the dump.
